ferrous-game/Assets/Scripts/Mechanics/SwitchManager.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Events;
+using System;
+
+namespace Ferrous.Mechanics
+{
+    public class SwitchManager : BaseSwitch
+    {
+        [SerializeField] private UnityEvent onAllSwitchesActivated;
+        [SerializeField] private List<BaseSwitch> activatedSwitches = new List<BaseSwitch>();
+        private bool allActivated = false;
```

Comment:
to be honest, I don't think you need this bool, why not just use `activated` as the condition and replace all instances of allActivated with activated?